File: GeneratedEntityReaderClasses/Components/TeleportComponent.cs
namespace GeneratedNoitaClasses;
public class TeleportComponent : NoitaComponentBase,  iNoitaType<TeleportComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaCVector2<NoitaFloat> target { get; set; } = new();
	public NoitaBool target_x_is_absolute_position { get; set; } = new();
	public NoitaBool target_y_is_absolute_position { get; set; } = new();
	public NoitaString source_particle_fx_file { get; set; } = new();
	public NoitaString target_particle_fx_file { get; set; } = new();
	public NoitaBool load_collapse_entity { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		target.Read(s);
		target_x_is_absolute_position.Read(s);
		target_y_is_absolute_position.Read(s);
		source_particle_fx_file.Read(s);
		target_particle_fx_file.Read(s);
		load_collapse_entity.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		target.Write(s);
		target_x_is_absolute_position.Write(s);
		target_y_is_absolute_position.Write(s);
		source_particle_fx_file.Write(s);
		target_particle_fx_file.Write(s);
		load_collapse_entity.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/TeleportComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class TeleportComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    target:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    target_x_is_absolute_position:NoitaBool = field(default_factory=NoitaBool)
    target_y_is_absolute_position:NoitaBool = field(default_factory=NoitaBool)
    source_particle_fx_file:NoitaString = field(default_factory=NoitaString)
    target_particle_fx_file:NoitaString = field(default_factory=NoitaString)
    load_collapse_entity:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.target.Read(s)
        self.target_x_is_absolute_position.Read(s)
        self.target_y_is_absolute_position.Read(s)
        self.source_particle_fx_file.Read(s)
        self.target_particle_fx_file.Read(s)
        self.load_collapse_entity.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.target.Write(s)
        self.target_x_is_absolute_position.Write(s)
        self.target_y_is_absolute_position.Write(s)
        self.source_particle_fx_file.Write(s)
        self.target_particle_fx_file.Write(s)
        self.load_collapse_entity.Write(s)

